src/webgpu/api/validation/capability_checks/limits/maxComputeWorkgroupsPerDimension.spec.ts

```diff
@@ -0,0 +1,92 @@
+import { keysOf } from '../../../../../common/util/data_tables.js';
+
+import { kLimitBaseParams, makeLimitTestGroup } from './limit_utils.js';
+
+const limit = 'maxComputeWorkgroupsPerDimension';
+export const { g, description } = makeLimitTestGroup(limit);
+
+const CreateComputePipelineTypes = {
+  createComputePipeline: true,
+  createComputePipelineAsync: true,
+};
+type CreateComputePipelineType = keyof typeof CreateComputePipelineTypes;
+const kCreateComputePipelineTypes = keysOf(CreateComputePipelineTypes);
+
+async function createComputePipeline(
+  device: GPUDevice,
+  descriptor: GPUComputePipelineDescriptor,
+  pipelineType: CreateComputePipelineType
+) {
+  switch (pipelineType) {
+    case 'createComputePipeline':
+      return device.createComputePipeline(descriptor);
+    case 'createComputePipelineAsync':
+      return await device.createComputePipelineAsync(descriptor);
+  }
+}
+
+g.test('createComputePipeline,async,at_over')
+  .desc(`Test using createComputePipeline,async at and over ${limit} limit`)
+  .params(
+    kLimitBaseParams.combine('pipelineType', kCreateComputePipelineTypes).combine('axis', [0, 1, 2])
+  )
+  .fn(async t => {
+    const { limitTest, testValueName, pipelineType, axis } = t.params;
+    await t.testDeviceWithRequestedLimits(
+      limitTest,
+      testValueName,
+      async ({ device, testValue, shouldError }) => {
+        const counts = [1, 1, 1];
+        counts[axis] = testValue;
+
+        const buffer = device.createBuffer({
+          size: 16,
+          usage: GPUBufferUsage.STORAGE,
+        });
+
+        const module = device.createShaderModule({
+          code: `
+          @group(0) @binding(0) var<storage, read_write> d: f32;
+          @compute @workgroup_size(1) fn main() {
+            d = 1.0;
+          }
+          `,
+        });
+
+        const pipeline = await createComputePipeline(
+          device,
+          {
+            layout: 'auto',
+            compute: {
+              module,
+              entryPoint: 'main',
+            },
+          },
+          pipelineType
+        );
+
+        const bindGroup = device.createBindGroup({
+          layout: pipeline.getBindGroupLayout(0),
+          entries: [
+            {
+              binding: 0,
+              resource: { buffer },
+            },
+          ],
+        });
+
+        const encoder = device.createCommandEncoder();
+        const pass = encoder.beginComputePass();
+        pass.setPipeline(pipeline);
+        pass.setBindGroup(0, bindGroup);
```

Comment:
I think you could technically do without the bind group, I don't think anything stipulates that compute shaders have any outputs.

Reply:
done